### DISCUSSION TITLE
EventHandler EventArgs consistency

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#7010] [General]
In the Avalonia repository when calling an EventHandler to which no argument needs to be passed there are two different conventions.

The classic one where EventArgs.Empty is used and the UWP one where null is used.

I think we should choose only one way.

### UWP Style
https://github.com/AvaloniaUI/Avalonia/blob/d3b21f589389b8bedfa75ed06d546658745e2089/src/Avalonia.Controls/SplitView.cs#L208

### Classical
https://github.com/AvaloniaUI/Avalonia/blob/ba8dcbe945affe397a86ddb4ded9fdbf667d1bda/src/Avalonia.Controls/TextBoxTextInputMethodClient.cs#L37

**Comment by robloo** [comment]
* Classic works with nullable reference types better. 
* UWP, in my opinion, shouldn't really be used as a baseline for decisions like this: it is sometimes quite ugly/hacky compared to WPF. Most of the time I assume the developers were less skilled than those for WPF and had no idea why things were really done the way they were.
* You also never know if UWP is just missing API's it shouldn't be (like Color.Empty) or if they are passing null to improve interop with native code (there are so many strange API decisions driven by the native/managed interop). https://github.com/microsoft/microsoft-ui-xaml/issues/2826#issuecomment-654519738 "We don't have support for members on structs in WinRT" illustrates my point but doesn't really apply here.
* Anyway, classic is better and will force all EventArgs to actually derive from the EventArgs base class as they should be anyway.
